crates/zir-codegen-cranelift/src/lib.rs

```diff
@@ -110,10 +110,268 @@ pub fn pointer_type(isa: &dyn TargetIsa) -> types::Type {
+        bb0.terminator = Some(Terminator {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: TerminatorKind::SwitchInt {
+                discr: Operand::Copy(Place::from_local(Local::new(3))),
+                targets: SwitchTargets::if_else(0, BasicBlock::new(2), BasicBlock::new(1)),
+            },
+        });
+        body.basic_blocks.push(bb0);
+
+        // bb1 (true branch): _0 = copy _1; goto -> bb3
+        let mut bb1 = BasicBlockData::new();
+        bb1.statements.push(Statement {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: StatementKind::Assign(
+                Place::from_local(Local::new(0)),
+                Rvalue::Use(Operand::Copy(Place::from_local(Local::new(1)))),
+            ),
+        });
+        bb1.terminator = Some(Terminator {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: TerminatorKind::Goto { target: BasicBlock::new(3) },
+        });
+        body.basic_blocks.push(bb1);
+
+        // bb2 (false branch): _0 = copy _2; goto -> bb3
+        let mut bb2 = BasicBlockData::new();
+        bb2.statements.push(Statement {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: StatementKind::Assign(
+                Place::from_local(Local::new(0)),
+                Rvalue::Use(Operand::Copy(Place::from_local(Local::new(2)))),
+            ),
+        });
+        bb2.terminator = Some(Terminator {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: TerminatorKind::Goto { target: BasicBlock::new(3) },
+        });
+        body.basic_blocks.push(bb2);
+
+        // bb3 (join point): return
+        let mut bb3 = BasicBlockData::new();
+        bb3.terminator = Some(Terminator {
+            source_info: SourceInfo { span: Span::DUMMY },
+            kind: TerminatorKind::Return,
+        });
+        body.basic_blocks.push(bb3);
+
+        body
+    }
+
+    /// Compiles and JIT-executes a function that takes no arguments and returns an i64.
+    fn jit_compile_and_run(body: &Body<'_>, name: &str) -> CodegenResult<i64> {
+        let module = create_jit_module()?;
+        let mut ctx = CodegenContext::new(module);
+
+        // Build signature for () -> i64
+        let mut sig = ctx.module.make_signature();
+        sig.returns.push(AbiParam::new(types::I64));
+
+        // Declare and define the function
+        let func_id = ctx.declare_function(name, sig.clone())?;
+        ctx.define_function(func_id, body, sig)?;
+
+        // Finalize and get function pointer
+        ctx.module
+            .finalize_definitions()
+            .map_err(|e| CodegenError::Module(format!("finalize error: {}", e)))?;
+
+        let code_ptr = ctx.module.get_finalized_function(func_id);
+
+        // Safety: We trust our codegen to produce valid code matching the signature
+        let func: fn() -> i64 = unsafe { std::mem::transmute(code_ptr) };
+        Ok(func())
+    }
```

Comment:
maybe its better to check compiled cranelift output instead of run it.
Develop a test solution that helps us declare a mir (in rust) and expect result of compilation clif or something other (research a production solution)